src/front/c/mod.rs

```diff
@@ -0,0 +1,668 @@
+                }
+            }
+            Expression::Cast(node) => {
+                let CastExpression {
+                    type_name,
+                    expression,
+                } = node.node;
+                let to_ty = s_type_(type_name.node.specifiers);
+                let expr = self.gen_expr(expression.node);
+                Ok(cast(to_ty, expr))
+            }
+            _ => unimplemented!("Expr {:#?} hasn't been implemented", expr),
+        };
+        self.unwrap(res)
+    }
+
+    fn gen_init(&mut self, derived_ty: Ty, init: Initializer) -> CTerm {
+        match init {
+            Initializer::Expression(e) => self.gen_expr(e.node),
+            Initializer::List(l) => {
+                // TODO: check length of values to initialized number
+                let mut values: Vec<CTerm> = Vec::new();
+                let inner_type = inner_ty(derived_ty);
+                for li in l {
+                    let expr = self.gen_init(inner_type.clone(), li.node.initializer.node.clone());
+                    values.push(expr)
+                }
+                let id = self
+                    .circ
+                    .zero_allocate(values.len(), 32, num_bits(inner_type.clone()));
+
+                for (i, v) in values.iter().enumerate() {
+                    let offset = bv_lit(i, 32);
+                    let v_ = v.term.term(&self.circ);
+                    self.circ.store(id, offset, v_);
+                }
+
+                CTerm {
+                    term: CTermData::CArray(inner_type, Some(id)),
+                    udef: false,
+                }
+            }
+        }
+    }
+
+    fn gen_decl(&mut self, decl: Declaration) -> CTerm {
+        let decl_info = get_decl_info(decl.clone());
+        let d = decl.declarators.first().unwrap().node.clone();
+        let base_ty: Ty = decl_info.ty;
+        let derived = &d.declarator.node.derived;
+        let derived_ty = self.derived_type_(base_ty, derived.to_vec());
+        let expr: CTerm;
+        if let Some(init) = d.initializer {
+            expr = self.gen_init(derived_ty.clone(), init.node);
+        } else {
+            expr = match derived_ty {
+                Ty::Array(size, ref ty) => {
+                    let id = self
+                        .circ
+                        .zero_allocate(size.unwrap(), 32, num_bits(*ty.clone()));
+                    CTerm {
+                        term: CTermData::CArray(*ty.clone(), Some(id)),
+                        udef: false,
+                    }
+                }
+                _ => derived_ty.default(),
+            }
+        }
+
+        let res = self.circ.declare_init(
+            decl_info.name,
+            derived_ty.clone(),
+            Val::Term(cast(Some(derived_ty), expr.clone())),
+        );
+        self.unwrap(res);
+        expr
+    }
+
+    fn get_const_iters(&mut self, for_stmt: ForStatement) -> Option<ConstIteration> {
```

Comment:
This function isn't quite right b/c the loop body could modify the iteration variable. We had a nice way to detect that in the Haskell compiler that would be a bit harder to replicate here. For now, why don't you write a comment noting the inaccuracy.